File: gcd.cpp
#include <iostream>
using namespace std;

long long gcd(int a,int b){
  long long r = a;
  if(b % a == 0){
    return r;
  }
  else{
    r = b % a;
    return gcd(r, a);
  }
}

int main(){
  int n1, n2;
  cout << "Enter n1 and n2 (seperate with space): ";
  cin >> n1 >> n2;
  long long gc = gcd(n1, n2);
  cout << gc << endl;
  return 0;
}

File: gcd.js
function gcd(a, b, debug = false){
	let r = a;
	if(b % a === 0){
		if(debug){
			console.log(`${b} / ${a} = ${(b - 0)/a} + ${0}`);
			console.log('------------------');
			console.log(`| [*] gcd = ${r}  |`);
			console.log('------------------');
		}
		return r;
	}
	r = b % a;
	if(debug) console.log(`${b} / ${a} = ${(b - r)/a} + ${r}`);
	return gcd(r, a, debug)
}

// console.log(gcd(100, 10, true)); // 10
